Source/ASDisplayNode.mm

```diff
@@ -828,6 +830,94 @@ - (void)nodeViewDidAddGestureRecognizer
   _flags.viewEverHadAGestureRecognizerAttached = YES;
 }
 
+#pragma mark UIResponder
+
+#define HANDLE_RESPONDER_METHOD(__sel) \
+  if (checkFlag(Synchronous)) { \
```

Comment:
This should probably assert main thread. At least -becomeFirstResponder might be reasonable for a developer to expect to be a UIViewBridge property and threadsafe to call, but this implementation triggers view creation.

Maybe worth even a special assertion in -becomeFirstResponder that tells developers to call that method on the main thread or in onDidLoad / -didLoad.

Reply:
Agree, I added a main thread assertion for all responder methods.